content/blog/2019-11-05-tidync.md

```diff
@@ -0,0 +1,1099 @@
+
+There is a tension between the **tidyverse** and **scientific array data** that comes down to this difference in data storage, and the intermediate forms used to get between one form and another. We also think this tension has been exaggerated in unproductive ways. 
+
+<br>
+
+### tidync
+
+The tidync package provides a compromise position, allowing efficient interaction with NetCDF files, producing native-array *or* tidy-data-frame output as desired. It delays any data-reading activity until after the output format is chosen. In particular, tidync exists in order to reduce the amount of plumbing code required to get to the data, and allows an interactive way to convert between the different spaces (coordinates and indices) in which the data can be referenced.  
+
+In pseudo-code, there are only a few simple steps, at each step we can save the result and explore a summary. 
+
+1. Connect to a data source and retrieve metadata, and read a summary: 
+
+```R
+src <- tidync(<netcdf-source>)
+print(src)
+```
+
+2. By default the largest array-space (grid) is *activated* and usually this will be the right choice - if required we can nominate a different grid using `activate()`. 
+
+```R
+src <- src %>% activate(<a different grid>)
+```
+
+3. Apply subsetting to *slice arrays* by coordinate or index, this step is optional but very important for large and complicated data sources. 
+
+```R
+## lazy subsetting by value or index
+src_slc <- src %>% hyper_filter(<filter expressions on dimensions>)
+```
+
+4. Finally, choose an output format - list of arrays, a data frame, or a `tbl_cube`. 
+
+```R
+src_slc %>% hyper_array()
+
+src_slc %>% hyper_tibble()
+
+src_slc %>% hyper_tbl_cube()
+```
+
+There are various other packages for NetCDF in R, the main ones being [RNetCDF](https://CRAN.r-project.org/package=RNetCDF) and [ncdf4](https://CRAN.r-project.org/package=ncdf4). These are both *lower-level* tools than tidync - they are interfaces to the underlying NetCDF library, and tidync uses both to read information and data. The [raster](https://CRAN.r-project.org/package=raster), and [stars](https://CRAN.r-project.org/package=stars) packages provide quite different approaches and stars is more general than raster, but is similarly *higher-level* than tidync. 
+
+To follow along with the code below requires all of the following packages, and we assume that recent versions are in use, particularly `ncmeta (>= 0.2.0)`, 
+`tidync (>= 0.2.2)`, and `tidyr (>= 1.0.0)`. 
+
+```R
+install.packages(c("ncmeta", "tidync", "maps", "stars", "ggplot2", "devtools", 
+                   "stars", "RNetCDF", "raster", "dplyr", "tidyr"))
+```
+
+The official website for tidync is https://docs.ropensci.org/tidync/ and the latest release can be found on [CRAN](https://CRAN.R-project.org/package=tidync). 
```

Comment:
maybe these links for where to find the pkg can be moved to the top of this section

Reply:
done